src/main/java/io/vertx/core/buffer/ByteBufUtils.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) 2011-2017 Contributors to the Eclipse Foundation
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ */
+
+package io.vertx.core.buffer;
+
+import io.netty.buffer.*;
+import io.netty.util.AsciiString;
+import io.netty.util.CharsetUtil;
+import io.netty.util.internal.PlatformDependent;
+
+import java.nio.charset.Charset;
+import java.util.Objects;
+
+import static io.netty.util.internal.ObjectUtil.checkPositiveOrZero;
+import static io.netty.util.internal.StringUtil.isSurrogate;
+
+public final class ByteBufUtils {
+  private ByteBufUtils() {
+
+  }
+
+  // this wrapper is used to save copying byte[]
+  private static final class UnpooledWrappedByteBuf extends UnpooledHeapByteBuf {
+
+    protected UnpooledWrappedByteBuf(ByteBufAllocator alloc, byte[] initialArray, int maxCapacity) {
+      super(alloc, initialArray, maxCapacity);
+    }
+  }
+
+  public static UnpooledHeapByteBuf unpooledNotInstrumentedHeapByteBuf(int initialCapacity, int maxCapacity) {
+    checkPositiveOrZero(initialCapacity, "initialCapacity");
+    // this save instrumented heap ByteBuf allocation that would save statistics collection
```

Comment:
why would we avoid statistics collection ?

Reply:
On massive scale benchs and many small sized temporary buffers, the statistics collection can cost more then the buffer allocation